configapi/api_inventory.go

```diff
@@ -93,39 +95,76 @@ func PostGnb(c *gin.Context) {
 // @Param        gnb-name    path    string    true    "Name of the gNB"
 // @Security     BearerAuth
 // @Success      200  {object}  nil  "gNB deleted"
-// @Failure      400  {object}  nil  "Failed to delete the gNB"
+// @Failure      400  {object}  nil  "Bad request"
 // @Failure      401  {object}  nil  "Authorization failed"
 // @Failure      403  {object}  nil  "Forbidden"
+// @Failure      500  {object}  nil  "Failed to delete gNB"
 // @Router       /config/v1/inventory/gnb/{gnb-name}  [delete]
 func DeleteGnb(c *gin.Context) {
+	logger.WebUILog.Infoln("received a DELETE gNB request")
 	setInventoryCorsHeader(c)
-	if err := handleDeleteGnb(c); err == nil {
-		c.JSON(http.StatusOK, gin.H{})
-	} else {
-		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+	gnbName, exists := c.Params.Get("gnb-name")
+	if !exists {
+		errorMessage := "delete gNB request is missing path param `gnb-name`"
+		logger.WebUILog.Errorln(errorMessage)
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorMessage})
+		return
+	}
+	filter := bson.M{"name": gnbName}
+	err := handleDeleteGnbTransaction(filter, gnbName)
+	if err != nil {
+		logger.WebUILog.Errorw("failed to delete gNB", "gnbName", gnbName, "error", err)
+		c.JSON(http.StatusInternalServerError, gin.H{"error": "failed to delete gNB"})
+		return
 	}
+	logger.WebUILog.Infof("successfully executed DELETE gNB %v request", gnbName)
+	c.JSON(http.StatusOK, gin.H{})
+}
+
+func handleDeleteGnbTransaction(filter bson.M, gnbName string) error {
```

Comment:
```suggestion
func handleDeleteGnbTransaction(ctx context.Context, filter bson.M, gnbName string) error {
```
The caller method can pass the gin.Context here.